# datos 2018

## tabla T_Vivienda
primary key: ID_VIV

In [1]:
from simpledbf import Dbf5
base_path="datos_2018/bd_envipe2018_dbf/BASE_DE_DATOS_ENVIPE_2018_dbf/"
dbf_tvivienda=Dbf5(base_path+"TVivienda.dbf")
tvivienda=dbf_tvivienda.to_dataframe()
print(tvivienda.shape)
tvivienda.head(3)

PyTables is not installed. No support for HDF output.
(89572, 20)


ID_VIV      UPM VIV_SEL AREAM CVE_ENT         NOM_ENT CVE_MUN  \
0  0100008.01  0100008      01    14      01  Aguascalientes     001   
1  0100008.02  0100008      02    14      01  Aguascalientes     001   
2  0100008.04  0100008      04    14      01  Aguascalientes     001   

          NOM_MUN RESULT_V AP1_1 AP1_2 AP2_1 AP2_2 AP2_3 NAP2_3 FAC_VIV  \
0  Aguascalientes       02     1     2    04     1   NaN     01  000226   
1  Aguascalientes       02     2     2    01     1   NaN     01  000226   
2  Aguascalientes       02     1     2    02     1   NaN     01  000226   

  FAC_VIV_AM DOMINIO EST_DIS UPM_DIS  
0     000226       U     003   00001  
1     000226       U     003   00001  
2     000226       U     003   00001

## tabla Thogar
primary key: ID_HOG

In [2]:
dbf_thogar=Dbf5(base_path+"THogar.dbf")
thogar=dbf_thogar.to_dataframe()
print(thogar.shape)
thogar.head(3)

(91664, 18)


ID_VIV         ID_HOG      UPM VIV_SEL HOGAR RESUL_H N_INF R_SEL  \
0  0100008.01  0100008.01.01  0100008      01    01       B    02    04   
1  0100008.02  0100008.02.01  0100008      02    01       B    01    01   
2  0100008.04  0100008.04.01  0100008      04    01       B    01    02   

  CVE_ENT         NOM_ENT CVE_MUN         NOM_MUN TOT_PER FAC_HOG FAC_HOG_AM  \
0      01  Aguascalientes     001  Aguascalientes      04  000226     000226   
1      01  Aguascalientes     001  Aguascalientes      01  000226     000226   
2      01  Aguascalientes     001  Aguascalientes      02  000226     000226   

  DOMINIO EST_DIS UPM_DIS  
0       U     003   00001  
1       U     003   00001  
2       U     003   00001

## tabla TSDem
primary key: ID_PER

In [3]:
dbf_tsdem=Dbf5(base_path+"TSDem.dbf")
tsdem=dbf_tsdem.to_dataframe()
print(tsdem.shape)
tsdem.head(3)

(326216, 26)


ID_VIV         ID_HOG            ID_PER      UPM VIV_SEL HOGAR RESUL_H  \
0  0100008.01  0100008.01.01  0100008.01.01.01  0100008      01    01       B   
1  0100008.01  0100008.01.01  0100008.01.01.02  0100008      01    01       B   
2  0100008.01  0100008.01.01  0100008.01.01.03  0100008      01    01       B   

  N_REN PAREN SEXO  ... CVE_MUN         NOM_MUN AP3_8 AP3_9 AP3_10 FAC_HOG  \
0    01     1    1  ...     001  Aguascalientes     6     4    NaN  000226   
1    02     2    2  ...     001  Aguascalientes     5     4    NaN  000226   
2    03     3    1  ...     001  Aguascalientes     1   NaN      2  000226   

  FAC_HOG_AM DOMINIO EST_DIS UPM_DIS  
0     000226       U     003   00001  
1     000226       U     003   00001  
2     000226       U     003   00001  

[3 rows x 26 columns]

 ## tabla TPer_Vic1
 Tabla de percepción sobre seguridad y desempeño institucional
 
 primary key: ID_PER

In [8]:
dbf_tper_vic1=Dbf5(base_path+"TPer_Vic1.dbf")
tper_vic1=dbf_tper_vic1.to_dataframe()
print(tper_vic1.shape)
tper_vic1.head(3)

(91541, 186)


ID_VIV         ID_HOG            ID_PER      UPM VIV_SEL HOGAR RESUL_H  \
0  0100008.01  0100008.01.01  0100008.01.01.04  0100008      01    01       B   
1  0100008.02  0100008.02.01  0100008.02.01.01  0100008      02    01       B   
2  0100008.04  0100008.04.01  0100008.04.01.02  0100008      04    01       B   

  R_SEL SEXO EDAD  ... AP5_6_10 AP5_8 FAC_HOG FAC_ELE FAC_HOG_AM FAC_ELE_AM  \
0    04    2   37  ...      NaN     3  000226  000903     000226     000903   
1    01    1   68  ...      NaN     9  000226  000226     000226     000226   
2    02    1   25  ...        2     3  000226  000452     000226     000452   

  DOMINIO ESTRATO EST_DIS UPM_DIS  
0       U       4     003   00001  
1       U       4     003   00001  
2       U       4     003   00001  

[3 rows x 186 columns]

## tabla TPer_Vic2
Tabla de percepción sobre seguridad, desempeño institucional y victimización
primary key: ID_PER

In [10]:
dbf_tper_vic2=Dbf5(base_path+"TPer_Vic2.dbf")
tper_vic2=dbf_tper_vic2.to_dataframe()
print(tper_vic2.shape)
tper_vic2.head(3)

(91541, 149)


ID_VIV         ID_HOG            ID_PER      UPM VIV_SEL HOGAR RESUL_H  \
0  0100008.01  0100008.01.01  0100008.01.01.04  0100008      01    01       B   
1  0100008.02  0100008.02.01  0100008.02.01.01  0100008      02    01       B   
2  0100008.04  0100008.04.01  0100008.04.01.02  0100008      04    01       B   

  R_SEL SEXO EDAD  ... AP7_3_15 AP7_4_15 FAC_HOG FAC_ELE FAC_HOG_AM  \
0    04    2   37  ...        2      NaN  000226  000903     000226   
1    01    1   68  ...        2      NaN  000226  000226     000226   
2    02    1   25  ...        2      NaN  000226  000452     000226   

  FAC_ELE_AM DOMINIO ESTRATO EST_DIS UPM_DIS  
0     000903       U       4     003   00001  
1     000226       U       4     003   00001  
2     000452       U       4     003   00001  

[3 rows x 149 columns]

## tabla TMod_Vic
Tabla del módulo sobre victimización
 
primary key: ID_DEL

In [13]:
dbf_tmod_vic=Dbf5(base_path+"TMod_Vic.dbf")
tmod_vic=dbf_tmod_vic.to_dataframe()
print(tmod_vic.shape)
tmod_vic.head(3)

(46674, 129)


ID_VIV         ID_HOG            ID_PER                 ID_DEL  \
0  0100721.03  0100721.03.01  0100721.03.01.03  0100721.03.01.03.02.2   
1  0100721.03  0100721.03.01  0100721.03.01.03  0100721.03.01.03.01.1   
2  0100721.03  0100721.03.01  0100721.03.01.03  0100721.03.01.03.02.3   

       UPM VIV_SEL HOGAR ND_TIPO TD_TIPO BPCOD  ... BP6_1 BP6_2 BP6_3 BP7_1  \
0  0100721      03    01       2       5    02  ...   NaN   NaN   NaN   NaN   
1  0100721      03    01       1       1    01  ...   NaN   NaN   NaN   NaN   
2  0100721      03    01       3       5    02  ...   NaN   NaN   NaN   NaN   

  FAC_DEL FAC_DEL_AM DOMINIO ESTRATO EST_DIS UPM_DIS  
0  000201     000201       U       3     002   00079  
1  000201     000201       U       3     002   00079  
2  000201     000201       U       3     002   00079  

[3 rows x 129 columns]

In [15]:
len( tmod_vic["ID_DEL"].unique() )

46674

### Probando los datos que vienen en la tabla tpervic1, tpervic2
saber cual es subconjunto de otras

In [18]:
merg1=tsdem[["ID_PER","EDAD","SEXO"]].merge(tper_vic1[["ID_PER","EDAD","SEXO"]],left_index=False,right_index=False,on="ID_PER",how="inner")
print(merg1.shape)
merg1.head()

(91541, 5)


ID_PER EDAD_x SEXO_x EDAD_y SEXO_y
0  0100008.01.01.04     37      2     37      2
1  0100008.02.01.01     68      1     68      1
2  0100008.04.01.02     25      1     25      1
3  0100008.05.01.01     56      2     56      2
4  0100022.02.01.01     41      2     41      2

In [19]:
merg2=merg1.merge(tper_vic1[["ID_PER","EDAD","SEXO"]],left_index=False,right_index=False,on="ID_PER",how="inner")
print(merg2.shape)
merg2.head()

(91541, 7)


ID_PER EDAD_x SEXO_x EDAD_y SEXO_y EDAD SEXO
0  0100008.01.01.04     37      2     37      2   37    2
1  0100008.02.01.01     68      1     68      1   68    1
2  0100008.04.01.02     25      1     25      1   25    1
3  0100008.05.01.01     56      2     56      2   56    2
4  0100022.02.01.01     41      2     41      2   41    2

**lo anterior indica indica que** la tabla tpervic1 y tepervic2 son un subconjunto de la tabla tsdem, es decir, 
que que de las 326 mil personas que contiene la tabla tsdem que podria interpretarse como el catalogo de personas consultadas,
solo 91541 personas contestaron la encuesta de persepcion 1 y la encuesta de percepcion2 

- y todas las que contestaron la de percepcion 1 son exactamente las mismas que contestaron la de percepcion2

### revisando los datos de tmod_vic
Se supone que son los delitos que ya les sucedieron a las personas encuestadas, haciendo un conteo agrupado por personas, 
se podra saber cuantas y cuales personas son las que estan contempladas en esta tabla

In [23]:
group_11=tmod_vic[["ID_PER","ID_DEL"]].groupby(by="ID_PER").count()
print(group_11.shape[0],"Es el numero de personas registradas en esta tabla")
group_11.head()

26572 Es el numero de personas registradas en esta tabla


ID_DEL
ID_PER                  
0100022.02.01.01       7
0100022.03.01.02       1
0100022.05.01.01       1
0100059.03.01.01       1
0100074.01.01.02       1

In [25]:
merg3=merg2.merge(group_11,left_index=False,right_index=True,left_on="ID_PER",how="inner")
print(merg3.shape)
merg3.head()

(26572, 8)


ID_PER EDAD_x SEXO_x EDAD_y SEXO_y EDAD SEXO  ID_DEL
4   0100022.02.01.01     41      2     41      2   41    2       7
5   0100022.03.01.02     39      1     39      1   39    1       1
7   0100022.05.01.01     70      2     70      2   70    2       1
9   0100059.03.01.01     27      1     27      1   27    1       1
11  0100074.01.01.02     23      2     23      2   23    2       1

**Lo anterior dice que** de las 91 541 personas que contestaron percepcion 1 y percepcion 2, hay 26572 personas que contestaron 
haber sufrido algun delito

In [32]:
tmod_vic.iloc[[0]]

ID_VIV         ID_HOG            ID_PER                 ID_DEL  \
0  0100721.03  0100721.03.01  0100721.03.01.03  0100721.03.01.03.02.2   

       UPM VIV_SEL HOGAR ND_TIPO TD_TIPO BPCOD  ... BP6_1 BP6_2 BP6_3 BP7_1  \
0  0100721      03    01       2       5    02  ...   NaN   NaN   NaN   NaN   

  FAC_DEL FAC_DEL_AM DOMINIO ESTRATO EST_DIS UPM_DIS  
0  000201     000201       U       3     002   00079  

[1 rows x 129 columns]

In [33]:
tmod_vic.shape

(46674, 129)

In [35]:
len(tmod_vic["ID_PER"].unique())

26572

In [36]:
len(tmod_vic["ID_DEL"].unique())

46674

In [37]:
tmod_vic["BP1_2C"]

Index(['ID_VIV', 'ID_HOG', 'ID_PER', 'ID_DEL', 'UPM', 'VIV_SEL', 'HOGAR',
       'ND_TIPO', 'TD_TIPO', 'BPCOD',
       ...
       'BP6_1', 'BP6_2', 'BP6_3', 'BP7_1', 'FAC_DEL', 'FAC_DEL_AM', 'DOMINIO',
       'ESTRATO', 'EST_DIS', 'UPM_DIS'],
      dtype='object', length=129)